Storage/container/one_to_one.h

```diff
@@ -0,0 +1,268 @@
+/*******************************************************************************
+The MIT License (MIT)
+
+Copyright (c) 2016 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+*******************************************************************************/
+
+#ifndef CURRENT_STORAGE_CONTAINER_ONE_TO_ONE_H
+#define CURRENT_STORAGE_CONTAINER_ONE_TO_ONE_H
+
+#include "common.h"
+#include "sfinae.h"
+
+#include "../base.h"
+
+#include "../../TypeSystem/optional.h"
+#include "../../Bricks/util/comparators.h"  // For `CurrentHashFunction`.
+
+namespace current {
+namespace storage {
+namespace container {
+
+template <typename T,
+          typename T_UPDATE_EVENT,
+          typename T_DELETE_EVENT,
+          template <typename...> class ROW_MAP,
+          template <typename...> class COL_MAP>
+class GenericOneToOne {
+ public:
+  using T_ROW = sfinae::ENTRY_ROW_TYPE<T>;
+  using T_COL = sfinae::ENTRY_COL_TYPE<T>;
+  using T_KEY = std::pair<T_ROW, T_COL>;
+  using T_ELEMENTS_MAP = std::unordered_map<T_KEY, std::unique_ptr<T>, CurrentHashFunction<T_KEY>>;
+  using T_FORWARD_MAP = ROW_MAP<T_ROW, const T*>;
+  using T_TRANSPOSED_MAP = COL_MAP<T_COL, const T*>;
+  using T_REST_BEHAVIOR = rest::behavior::Matrix;
+
+  explicit GenericOneToOne(MutationJournal& journal) : journal_(journal) {}
+
+  bool Empty() const { return map_.empty(); }
+  size_t Size() const { return map_.size(); }
+
+  // Adds specified object, removes and/or overwrites all existing entries
```

Comment:
Why not write two or three sentences describing the behavior a bit more clearly? `removes and/or overwrites` - so it removes, or overwrites, or both? Does it depend on something? Short comments are awesome, but let's not sacrifice clarity.